repose-aggregator/components/filters/openstack-identity-v3-filter/src/test/scala/org/openrepose/filters/openstackidentityv3/config/OpenStackIdentityV3SchemaTest.scala

```diff
@@ -32,53 +33,151 @@ class OpenStackIdentityV3SchemaTest extends ConfigurationTest {
+          |    <pre-authorized-roles>
+          |        <role>Role</role>
+          |    </pre-authorized-roles>
+          |</openstack-identity-v3>""".stripMargin
       intercept[SAXParseException] {
         validator.validateConfigString(config)
-      }.getLocalizedMessage should include ("Cannot define the deprecated roles-which-bypass-project-id-check and pre-authorized-roles")
+      }.getLocalizedMessage should include("Cannot define the deprecated roles-which-bypass-project-id-check and pre-authorized-roles")
+    }
+
+    cacheTimeouts.foreach { cacheTimeout =>
+      it(s"should successfully validate with a cache timeout $cacheTimeout attribute") {
+        val config =
+          s"""<openstack-identity-v3 xmlns="http://docs.openrepose.org/repose/openstack-identity-v3/v1.0"
+             |                       connection-pool-id="identity-v3-pool">
+             |    <openstack-identity-service
+             |            username="admin_username"
+             |            password="admin_password"
+             |            uri="http://identity.example.com"/>
+             |    <cache>
+             |        <timeouts $cacheTimeout="300"/>
+             |    </cache>
+             |</openstack-identity-v3>""".stripMargin
+        validator.validateConfigString(config)
+      }
+    }
+  }
+
+  private val coreSpringProvider = CoreSpringProvider.getInstance()
+  coreSpringProvider.initializeCoreContext("/etc/repose", false)
+  private val reposeVersion = coreSpringProvider.getCoreContext.getEnvironment.getProperty(
+    ReposeSpringProperties.stripSpringValueStupidity(ReposeSpringProperties.CORE.REPOSE_VERSION))
+  if (reposeVersion.startsWith("9.")) {
+    describe("deprecated schema validation") {
+      cacheTimeouts.foreach { cacheTimeout =>
+        it(s"should fail to validate when both a deprecated $cacheTimeout element and new attribute are defined") {
+          val config =
+            s"""<openstack-identity-v3 xmlns="http://docs.openrepose.org/repose/openstack-identity-v3/v1.0"
+               |                       connection-pool-id="identity-v3-pool">
+               |    <openstack-identity-service
+               |            username="admin_username"
+               |            password="admin_password"
+               |            uri="http://identity.example.com"/>
+               |    <cache>
+               |        <timeouts $cacheTimeout="300">
+               |            <$cacheTimeout>300</$cacheTimeout>
+               |        </timeouts>
+               |    </cache>
+               |</openstack-identity-v3>""".stripMargin
+          val
+          exception = intercept[SAXParseException] {
+            validator.validateConfigString(config)
+          }
+          exception.getLocalizedMessage should include(s"Cannot define both a deprecated $cacheTimeout element and the new $cacheTimeout attribute")
+        }
+      }
+
+      cacheTimeouts.permutations.map(_.take(2)) foreach { timeouts =>
```

Comment:
Why the `_.take(2)` here? It doesn't seem necessary?

Reply:
Also, for the sake of consistency, maybe make the ` foreach {` call `.foreach {`.